### ISSUE TITLE
Cannot restore snapshot for particular machine in conjunction with --provision-with option

### ISSUE THREAD

**Issue by zofrex** [hashicorp/vagrant#8994]
### Vagrant version
2.0.0

### Host operating system
macOS Sierra 10.12.6

### Guest operating system
N/A

### Vagrantfile
N/A

### Debug output
N/A

### Expected behavior
Restore the snapshot "preupgrade" for the machine "webserver"

### Actual behavior
Complained that no machine named "preupgrade" exists:

```
The machine with the name 'preupgrade' was not found configured for
this Vagrant environment.
```

### Steps to reproduce

`$ vagrant snapshot restore --provision-with sync-clock webserver preupgrade`

Where "webserver" is the name of a Vagrant machine and "preupgrade" is the name of a snapshot:

```
$ vagrant snapshot list webserver
preupgrade
```

In case the order is wrong I also tried the following:

`vagrant snapshot restore --provision-with sync-clock preupgrade webserver`

`vagrant snapshot restore --provision-with sync-clock preupgrade`

All have the same result.

**Comment by stonepresto** [comment]
## Vagrant version

2.1.2

## Host OS

Windows 10 1809

## Steps to reproduce

```
vagrant up --provision-with <provisioner1>
vagrant snapshot save <machine> <snapshot>
vagrant snapshot restore <machine> <snapshot> --provision-with <provisioner2>
```

## Expected result

`provisioner1` is ran the first time, and upon restoring the snapshot `provisioner2` is ran but not `provisioner1`

## Actual result

Upon restoring the snapshot, I recieve `The machine with the name <snapshot> was not found configured for this Vagrant environment.`

**Comment by chrisroberts** [comment]
Fixed via #10490
